librosa/filters.py

```diff
@@ -759,25 +762,18 @@ def wavelet_lengths(
             f"Frequency array={freqs} must be in strictly ascending order"
         )
 
-    # Filters have ~1bin bandwidth centered at the current frequency
-    #
-    # For each frequency, we compute alpha by the separation between
-    # the geometric mean of the frequency with its neighbors:
-    #
-    # sqrt(freqs[k+1] * freqs[k]) - sqrt(freqs[k] * freqs[k-1])
-    #
-    # At the boundaries, k=0 and -1, we geometrically reflect
-    if len(freqs) > 1:
-        alpha = np.empty(len(freqs))
-        # If we have at least two frequencies, we can infer bandwidth
-        freq_roots = freqs ** 0.5
-        alpha[1:-1] = (freq_roots[2:] - freq_roots[:-2]) / freq_roots[1:-1]
-
-        freq_low = freqs[0] / freq_roots[1]
-        alpha[0] = (freq_roots[1] - freq_low) / freq_roots[0]
-
-        freq_high = freqs[-1] / freq_roots[-2]
-        alpha[-1] = (freq_high - freq_roots[-2]) / freq_roots[-1]
+    # We need at least 4 frequencies to infer alpha
+    if len(freqs) > 4:
+        # Approximate the local octave resolution
+        bpo = np.empty(len(freqs))
+        logf = np.log2(freqs)
+        bpo[0] = 2/(logf[2] - logf[0])
```

Comment:
any way we could pass `bins_per_octave` to this function so as to use octave equivalence for local BPO estimation?

```
bpo[2:-2] = 4/(logf[4:] - logf[:-4])
bpo[:2] = bpo[bins_per_octave:(bins_per_octave+2)]
bpo[-2:] = bpo[-(bins_per_octave+2):-bins_per_octave]
```

where `bins_per_octave` is the user-provided constant

Reply:
if you're unhappy with the `len(freqs)>4` requirement, another way to estimate the local `bpo` is:

```
bpo[1:-1] = 2/(logf[2:] - logf[:-2])
```

now we have a `len(freqs)>2` requirement

Reply:
> any way we could pass bins_per_octave to this function so as to use octave equivalence for local BPO estimation?

Eh... we could, but it seems like it's inching toward API bloat.  We already pass through a backup alpha parameter to cover this case, and I don't think we need both.  Maybe BPO is better than alpha?

> if you're unhappy with the len(freqs)>4 requirement, another way to estimate the local bpo is:

I like that; it does seem a bit more direct to me.  But does it satisfy the conditions you had before?

Reply:
if we work with evenly spaced filters, then `bpo` will be a constant array, so whether we estimate it from second-order differences or fourth-order differences (or indeed, differences of any order) will not change the result: it will remain correct.

> But does it satisfy the conditions you had before?

yes.

Reply:
```
logf[n] = logf0 + n/B
logf[n+k] - logf[n-k] = (n+k)/B - (n-k)/B = 2k/B
2k/(logf[n+k] - logf[n-k]) = B
```

set `k=1` for second-order differences. `k=2` for fourth-order differences. the result is always `B`.
in light of this, i think that `k=1` should suffice

Reply:
Yeah, I'd agree with that.  Even if the filter spacing is not uniform, I expect that it would be preferable to have a more local estimate here anyway given the intended use in controlling bandwidth.

Reply:
> Eh... we could, but it seems like it's inching toward API bloat.

As i said offline, another option is to "discover" octave equivalence by seeking `(n, B)` such that `logf[n]+logf[n+B]=1` and work from there. No extra kwarg needed

Reply:
here's how to discover `bins_per_octave`

```python
bins_per_octave = 1 + np.argmax(np.real(np.fft.ifft(np.abs(np.fft.fft(np.modf(logf)[0])))**2)[1:])
```